File: content/python/templates/search-linear.py
# Linear Search

def linear_search(lst, target):
    """Search for target in list, return index or -1."""
    for i, item in enumerate(lst):
        if item == target:
            return i
    return -1

# Test
numbers = [3, 7, 2, 9, 1, 5]
target = 9

index = linear_search(numbers, target)
if index != -1:
    print(f"Found {target} at index {index}")
else:
    print(f"{target} not found")

File: content/cpp/templates/search-linear.cpp
// Linear Search

#include <iostream>
using namespace std;

int linearSearch(int arr[], int size, int target) {
    for (int i = 0; i < size; i++) {
        if (arr[i] == target) {
            return i;
        }
    }
    return -1;
}

int main() {
    int numbers[] = {3, 7, 2, 9, 1, 5};
    int size = sizeof(numbers) / sizeof(numbers[0]);
    int target = 9;
    
    int index = linearSearch(numbers, size, target);
    
    if (index != -1) {
        cout << "Found " << target << " at index " << index << endl;
    } else {
        cout << target << " not found" << endl;
    }
    
    return 0;
}
